src/main/java/cz/muni/fi/pv286/mmt/representation/BitParser.java

```diff
@@ -0,0 +1,141 @@
+package cz.muni.fi.pv286.mmt.representation;
+
+import cz.muni.fi.pv286.mmt.exceptions.InvalidBitCharacterException;
+import cz.muni.fi.pv286.mmt.exceptions.InvalidBitInputException;
+import cz.muni.fi.pv286.mmt.model.FromToOption;
+import cz.muni.fi.pv286.mmt.model.IoOption;
+import cz.muni.fi.pv286.mmt.model.Options;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Optional;
+
+/**
+ * Bit parser.
+ */
+public class BitParser extends RepresentationParser {
+
+    public BitParser(Options options) {
+        super(options);
+    }
+
+    /**
+     * Static method to allow padding of bytes from left or right.
+     */
+    public static byte[] padOutput(Optional<IoOption> inputFromToOption, byte[] bits) {
```

Comment:
**question:** Why is inputFromToOption as an option? Why the other arguments aren't optional?

*node:* This doesn't make any sense. Occurred at the other functions.